src/Api/SecretsManager/Controllers/AccessPoliciesController.cs

```diff
@@ -299,4 +301,34 @@ private async Task<(AccessClientType AccessClientType, Guid UserId)> GetAccessCl
         var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
         return (accessClient, userId);
     }
+
+    private async Task<List<BaseAccessPolicy>> SetCurrentUserInGroupAsync(
+        List<BaseAccessPolicy> accessPolicies, Guid organizationId)
+    {
+        if (!accessPolicies.OfType<GroupServiceAccountAccessPolicy>().Any()
+            && !accessPolicies.OfType<GroupProjectAccessPolicy>().Any())
+        {
+            return accessPolicies;
+        }
+
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgUsers = await _organizationUserRepository.GetManyByUserAsync(userId);
+        var orgUser = orgUsers.First(x => x.OrganizationId == organizationId);
+        var usersGroups = await _groupRepository.GetManyIdsByUserIdAsync(orgUser.Id);
```

Comment:
This is a N+1 query. If we want to take this approach we should create a new sproc for `GroupUser` which reads by the `userId`.